File: freight-triage-polyglot-ledger-recovery/environment/native/src/norm/dash_to_underscore.cpp
#include "freight/normalize.h"

#include <algorithm>

namespace freight {

// Normalizer: dash to underscore.
std::string norm_dash_to_underscore(const std::string& text) {
  std::string out(text);
  for (size_t i = 0; i < out.size(); ++i) {
    if (out[i] == '-') {
      out[i] = '_';
    }
  }
  return out;
}

}  // namespace freight

File: freight-triage-polyglot-ledger-recovery/environment/intake/src/com/freight/norm/DashToUnderscore.java
package com.freight.norm;

/** Normalizer: dash to underscore. */
public final class DashToUnderscore implements Normalizer {

    @Override
    public String name() {
        return "dash_to_underscore";
    }

    @Override
    public String apply(String text) {
        return text.replace('-', '_');
    }
}
